src/templates/layouts/event/index.tsx

```diff
@@ -0,0 +1,268 @@
+import React, { useState } from 'react'
+import { NodeEvent } from '@/types/dataTypes/drupal/node'
+import {
+  deriveMostRecentDate,
+  formatDateObject,
+  deriveFormattedTimestamp,
+  isEventInPast,
+} from './util'
+import { ContentFooter } from '@/templates/common/contentFooter'
+import { SocialLinks } from '../../common/socialLinks'
+import { MediaImage } from '@/templates/common/mediaImage'
+import { GoogleMapsDirections } from '@/templates/common/googleMapsDirections'
+import { recordEvent } from '@/lib/analytics/recordEvent'
+import { Event as FormattedEvent } from '@/types/dataTypes/formatted/event'
+
+export const Event = ({
+  title,
+  image,
+  description,
+  datetimeRange,
+  locationHumanReadable,
+  facilityLocation,
+  locationType,
+  urlOfOnlineEvent,
+  address,
+  cost,
+  socialLinks,
+  link,
+  additionalInfo,
+  eventCTA,
+  body,
+  listing,
+}: FormattedEvent) => {
+  const [showRecurringEvents, setShowRecurringEvents] = useState(false)
+
+  const toggleRecurringEvents = () => {
+    setShowRecurringEvents((prevState) => !prevState)
+  }
+  const formattedDates = formatDateObject(datetimeRange)
+
+  const mostRecentDate = deriveMostRecentDate(formattedDates)
+  const formattedTimestamp = deriveFormattedTimestamp(mostRecentDate)
```

Comment:
I wonder if these should be handled inside the query formatter itself. fine here for now, just a thought